src/sparsezoo/model/utils.py

```diff
@@ -122,24 +111,57 @@ def load_files_from_stub(
     params = None
     if isinstance(stub, str):
         stub, params = parse_zoo_stub(stub=stub, valid_params=valid_params)
-    _LOGGER.debug(f"load_files_from_stub: loading files from {stub}")
-    response = download_get_request(
-        args=stub,
-        force_token_refresh=force_token_refresh,
+    _LOGGER.debug(f"load_files_from_stub: calling  files from {stub}")
+
+    arguments = get_model_metadata_from_stub(stub)
+
+    api = GraphQLAPI()
+
+    models = api.fetch(
+        operation_body="models",
+        arguments=arguments,
+        fields=["modelId", "modelOnnxSizeCompressedBytes"],
     )
 
-    # piece of code required for backwards compatibility
-    model_response = response.get("model", {})
-    files = model_response.get("files", [])
-    files = restructure_request_json(request_json=files)
-    compressed_file_size = _get_compressed_size(files=files)
-    model_id = model_response.get("model_id")
-    if params is not None:
-        files = filter_files(files=files, params=params)
+    if len(models):
+        model_id = models[0]["model_id"]
+
+        files = api.fetch(
+            operation_body="files",
+            arguments={"model_id": model_id},
+        )
+        include_file_download_url(files)
+        files = restructure_request_json(request_json=files)
+
+        if params is not None:
+            files = filter_files(files=files, params=params)
+
+        training_results = api.fetch(
+            operation_body="training_results",
+            arguments={"model_id": model_id},
+        )
+
+        benchmark_results = api.fetch(
+            operation_body="benchmark_results",
+            arguments={"model_id": model_id},
+        )
```

Comment:
Is there any way to combine these three fetches into one to minimize http requests (& idk if each one makes a DB query on the backend, but it'd be good to reduce those for speed)?

Reply:
Yes good comment. There there is a future PR, blocked on this pr to implement that